### ISSUE TITLE
Foreign Key Constraints Not Scripted with DdlGeneratorFactory.TableGenerator

### ISSUE THREAD

**Issue by whibdev** [martinjw/dbschemareader#169]
When using the DdlGeneratorFactory.TableGenerator to write individual table scripts, foreign key constraints are not scripted.

> ORIGINAL TABLE CREATION SCRIPT
```
CREATE TABLE dbo.City
(
    Id			INT NOT NULL IDENTITY PRIMARY KEY,
    Name		NVARCHAR(50) NOT NULL
)
GO

CREATE TABLE dbo.Major
(
    Id			INT NOT NULL IDENTITY PRIMARY KEY,
    Name		NVARCHAR(50) NOT NULL
)
GO

CREATE TABLE dbo.Student
(
    Id			INT NOT NULL IDENTITY PRIMARY KEY,
    Name		NVARCHAR(50) NOT NULL,
    CityId		INT NOT NULL FOREIGN KEY REFERENCES dbo.City (Id) ON DELETE CASCADE,
    MajorId	INT NOT NULL FOREIGN KEY REFERENCES dbo.Major (Id) ON DELETE CASCADE
)
GO
```
> C# SCRIPT TO WRITE TABLE SCRIPTS
```
private static void PrintTableScripts()
{
    var sqlConn = new SqlConnection(connectionString);
    var dbReader = new DatabaseSchemaReader.DatabaseReader(sqlConn);
    var schema = dbReader.ReadAll();
    var factory = new DdlGeneratorFactory(SqlType.SqlServer);
    factory.UseGranularBatching = true;

    foreach (var t in schema.Tables)
    {
        var gen = factory.TableGenerator(t);
        Console.WriteLine(gen.Write());
    }
}
```

> OUTPUT (Missing Foreign Key Constraints)
```
CREATE TABLE [dbo].[City]
(
  [Id] INT IDENTITY(1,1) NOT NULL,
  [Name] NVARCHAR (50)  NOT NULL
)
GO
ALTER TABLE [dbo].[City] ADD CONSTRAINT [PK__City__3214EC07B47A945A] PRIMARY KEY ([Id]);

CREATE TABLE [dbo].[Major]
(
  [Id] INT IDENTITY(1,1) NOT NULL,
  [Name] NVARCHAR (50)  NOT NULL
)
GO
ALTER TABLE [dbo].[Major] ADD CONSTRAINT [PK__Major__3214EC07CFD450F8] PRIMARY KEY ([Id]);

CREATE TABLE [dbo].[Student]
(
  [Id] INT IDENTITY(1,1) NOT NULL,
  [Name] NVARCHAR (50)  NOT NULL,
  [CityId] INT  NOT NULL,
  [MajorId] INT  NOT NULL
)
GO
ALTER TABLE [dbo].[Student] ADD CONSTRAINT [PK__Student__3214EC07031CD5E9] PRIMARY KEY ([Id]);
```

> ADDITIONAL NOTES:
- The DdlGeneratorFactory.AllTablesGenerator scripts the foreign keys
- 'GO' termination statement is missing after the ALTER statements

**Comment by martinjw** [comment]
TableGenerator doesn't create the foreign keys because you can't be certain of the order that tables are created. In AllTablesGenerator you can- but not for an individual table. 

You can add the foreign keys with the constraint writer as a separate step.
```c#
//or just the same factoyr
var ddlGeneratorFactory = new DdlGeneratorFactory(sqlType);
var cw = ddlGeneratorFactory.ConstraintWriter(table);
var sql = cw.WriteForeignKeys();
```

I'm woring on a new version which adds some of this as extension methods, making it more discoverable
```c#
var createTable = table.ToSqlCreateTable();
var fks = table.ToSqlCreateForeignKeys();
```
There are overloads which allow you to set Granular Batching etc.

The missing GO after ALTER is a bug., which will be fixed in the new version.

**Comment by whibdev** [comment]
I like the idea of extension methods in the new version.

ConstraintWriter method is not accessible because it is internal ; can it be updated to public?
`var cw = ddlGeneratorFactory.ConstraintWriter(table);`

**Comment by martinjw** [comment]
New version released, 2.9.0. 

For now constraint writer is still internal - I think I wanted to limit the api to the migrationGenerator. But this extension method should expose what you need.

Here's the documentation

https://github.com/martinjw/dbschemareader/wiki/SQL-Generation/#tosql-extensions

**Comment by whibdev** [comment]
This works great 👍
